fail to build payment service image using Apple Silicon Mac

# Bug Report

Which version of the demo you are using?
current latest master ( https://github.com/open-telemetry/opentelemetry-demo-webstore/commit/48888b031cf16a9c56219720194c7aa8be10be85 )

## Symptom
**What is the expected behavior?**

payment service image should be built successfully.

**What is the actual behavior?**

an error comes up when building payment service image.

<details>

<summary>error details</summary>

```
 > [ghcr.io/open-telemetry/demo:0.1.0-paymentservice build 4/4] RUN npm ci --omit=dev:
#0 5.259 npm WARN deprecated @types/bson@4.2.0: This is a stub types definition. bson provides its own type definitions, so you do not need this installed.
#0 7.097 npm WARN deprecated grpc@1.24.11: This library will not receive further updates other than security fixes. We recommend using @grpc/grpc-js instead.
#0 12.23 npm notice
#0 12.23 npm notice New minor version of npm available! 8.11.0 -> 8.14.0
#0 12.23 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.14.0>
#0 12.23 npm notice Run `npm install -g npm@8.14.0` to update!
#0 12.23 npm notice
#0 12.24 npm ERR! code 1
#0 12.24 npm ERR! path /usr/src/app/node_modules/grpc
#0 12.24 npm ERR! command failed
#0 12.24 npm ERR! command sh -c node-pre-gyp install --fallback-to-build --library=static_library
#0 12.24 npm ERR! Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --library=static_library --module=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl/grpc_node.node --module_name=grpc_node --module_path=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
#0 12.24 npm ERR! node-pre-gyp info it worked if it ends with ok
#0 12.24 npm ERR! node-pre-gyp info using node-pre-gyp@1.0.9
#0 12.24 npm ERR! node-pre-gyp info using node@16.16.0 | linux | arm64
#0 12.24 npm ERR! node-pre-gyp info check checked for "/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl/grpc_node.node" (not found)
#0 12.24 npm ERR! node-pre-gyp http GET https://node-precompiled-binaries.grpc.io/grpc/v1.24.11/node-v93-linux-arm64-musl.tar.gz
#0 12.24 npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://node-precompiled-binaries.grpc.io/grpc/v1.24.11/node-v93-linux-arm64-musl.tar.gz
#0 12.24 npm ERR! node-pre-gyp WARN Pre-built binaries not installable for grpc@1.24.11 and node@16.16.0 (node-v93 ABI, musl) (falling back to source compile with node-gyp)
#0 12.24 npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://node-precompiled-binaries.grpc.io/grpc/v1.24.11/node-v93-linux-arm64-musl.tar.gz
#0 12.24 npm ERR! gyp info it worked if it ends with ok
#0 12.24 npm ERR! gyp info using node-gyp@9.0.0
#0 12.24 npm ERR! gyp info using node@16.16.0 | linux | arm64
#0 12.24 npm ERR! gyp info ok
#0 12.24 npm ERR! gyp info it worked if it ends with ok
#0 12.24 npm ERR! gyp info using node-gyp@9.0.0
#0 12.24 npm ERR! gyp info using node@16.16.0 | linux | arm64
#0 12.24 npm ERR! gyp ERR! find Python
#0 12.24 npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
#0 12.24 npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
#0 12.24 npm ERR! gyp ERR! find Python checking if "python3" can be used
#0 12.24 npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error
#0 12.24 npm ERR! gyp ERR! find Python checking if "python" can be used
#0 12.24 npm ERR! gyp ERR! find Python - "python" is not in PATH or produced an error
#0 12.24 npm ERR! gyp ERR! find Python
#0 12.24 npm ERR! gyp ERR! find Python **********************************************************
#0 12.24 npm ERR! gyp ERR! find Python You need to install the latest version of Python.
#0 12.24 npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#0 12.24 npm ERR! gyp ERR! find Python you can try one of the following options:
#0 12.24 npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#0 12.24 npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)
#0 12.24 npm ERR! gyp ERR! find Python - Set the environment variable PYTHON
#0 12.24 npm ERR! gyp ERR! find Python - Set the npm configuration variable python:
#0 12.24 npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#0 12.24 npm ERR! gyp ERR! find Python For more information consult the documentation at:
#0 12.24 npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#0 12.24 npm ERR! gyp ERR! find Python **********************************************************
#0 12.24 npm ERR! gyp ERR! find Python
#0 12.24 npm ERR! gyp ERR! configure error
#0 12.24 npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
#0 12.24 npm ERR! gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#0 12.24 npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#0 12.24 npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#0 12.24 npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#0 12.24 npm ERR! gyp ERR! stack     at exithandler (node:child_process:406:5)
#0 12.24 npm ERR! gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:418:5)
#0 12.24 npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
#0 12.24 npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
#0 12.24 npm ERR! gyp ERR! stack     at onErrorNT (node:internal/child_process:478:16)
#0 12.24 npm ERR! gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
#0 12.24 npm ERR! gyp ERR! System Linux 5.10.104-linuxkit
#0 12.24 npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--library=static_library" "--module=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl/grpc_node.node" "--module_name=grpc_node" "--module_path=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v93"
#0 12.24 npm ERR! gyp ERR! cwd /usr/src/app/node_modules/grpc
#0 12.24 npm ERR! gyp ERR! node -v v16.16.0
#0 12.24 npm ERR! gyp ERR! node-gyp -v v9.0.0
#0 12.24 npm ERR! gyp ERR! not ok
#0 12.24 npm ERR! node-pre-gyp ERR! build error
#0 12.24 npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --library=static_library --module=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl/grpc_node.node --module_name=grpc_node --module_path=/usr/src/app/node_modules/grpc/src/node/extension_binary/node-v93-linux-arm64-musl --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
#0 12.24 npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/src/app/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
#0 12.24 npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
#0 12.24 npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1092:16)
#0 12.24 npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
#0 12.24 npm ERR! node-pre-gyp ERR! System Linux 5.10.104-linuxkit
#0 12.24 npm ERR! node-pre-gyp ERR! command "/usr/local/bin/node" "/usr/src/app/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
#0 12.24 npm ERR! node-pre-gyp ERR! cwd /usr/src/app/node_modules/grpc
#0 12.24 npm ERR! node-pre-gyp ERR! node -v v16.16.0
#0 12.24 npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.9
#0 12.24 npm ERR! node-pre-gyp ERR! not ok
#0 12.24
#0 12.24 npm ERR! A complete log of this run can be found in:
#0 12.24 npm ERR!     /root/.npm/_logs/2022-07-16T12_44_43_079Z-debug-0.log
------
failed to solve: executor failed running [/bin/sh -c npm ci --omit=dev]: exit code: 1
```

</details>

## Reproduce

- execute the following command using Apple Silicon Mac or other arm64 machine

`docker compose build` or `docker compose up`

## Possible Solution

seems that `node:16-alpine` arm64 image doesn't have an executable python ( as a similar case:  https://github.com/docker/docker.github.io/issues/14547 ).
adding commands to install some dependencies to `src/paymentservice/Dockerfile` works fine.

```
diff --git a/src/paymentservice/Dockerfile b/src/paymentservice/Dockerfile
index 4850b66..f603e66 100644
--- a/src/paymentservice/Dockerfile
+++ b/src/paymentservice/Dockerfile
@@ -18,7 +18,7 @@ WORKDIR /usr/src/app/

 COPY ./src/paymentservice/package*.json ./

-RUN npm ci --omit=dev
+RUN apk add --no-cache python3 make g++ && npm ci --omit=dev
```

--- a/src/paymentservice/Dockerfile
+++ b/src/paymentservice/Dockerfile
@@ -18,7 +18,7 @@ WORKDIR /usr/src/app/
 
 COPY ./src/paymentservice/package*.json ./
 
-RUN npm ci --omit=dev
+RUN apk add --no-cache python3 make g++ && npm ci --omit=dev
 
 # -----------------------------------------------------------------------------
 

